nativelib/src/main/resources/scala-native/gc/shared/ThreadUtil.c

```diff
@@ -2,6 +2,20 @@
 #include <stdio.h>
 #include <limits.h>
 
+#if !(defined SCALANATIVE_GC_COMMIX ||                                         \
+      defined SCALANATIVE_MULTITHREADING_ENABLED)
```

Comment:
**thought:** it'd be good to have a single definition that we can check, e.g. "has pthread" - does something like this exist already?

Reply:
also, is this fine or should I flip the boolean (concurrent case first, fallback second)?

Reply:
By default, all the `shared` code should be compiled so something is not correct 100%. The `SCALANATIVE_GC_COMMIX` is defined only when selecting Commix so that code will get compiled and similar for other GCs. Does this help at all?

Reply:
It is used by all the internal GCs and also single threaded runtime because no guard against.

Reply:
hmm yeah. What I did here was inspired by [Wojciech's comment on the ticket](https://github.com/scala-native/scala-native/issues/3877#issuecomment-2060536437), but I see that this might not be the best way to do it because it's not really "shared" anymore.